src/base/bittorrent/sessionimpl.cpp

```diff
@@ -300,6 +300,32 @@ namespace
 #endif
 }
 
+class SessionImpl::Statistics final : public QObject
```

Comment:
Or maybe it's still worth going further and dropping this class altogether, putting all the logic directly into SessionImpl? Then we could also get rid of `getAlltimeDL()`/`getAlltimeUL()` methods of `Session`, and instead add the appropriate fields to `SessionStatus` for consistency.

Reply:
>Or maybe it's still worth going further and dropping this class altogether, putting all the logic directly into SessionImpl?

I'm ok with it.

Reply:
> > Or maybe it's still worth going further and dropping this class altogether, putting all the logic directly into SessionImpl?
> 
> I'm ok with it.

Done!